keras/src/models/model_test.py

```diff
@@ -944,3 +992,303 @@ def test_layers_setter(self):
             AttributeError, "`Model.layers` attribute is reserved"
         ):
             model.layers = [layers.Dense(4)]
+
+    def get_struct_loss(self, structure):
+        def loss_fn(y_true, y_pred):
+            tree.assert_same_structure(structure, y_true, check_types=False)
+            tree.assert_same_structure(structure, y_pred, check_types=False)
+            tree.map_structure(
+                lambda spec, tensor: self.assertEqual(spec.ndim, tensor.ndim),
+                structure,
+                y_true,
+            )
+            tree.map_structure(
+                lambda spec, tensor: self.assertEqual(spec.ndim, tensor.ndim),
+                structure,
+                y_pred,
+            )
+            flat_y_pred, flat_y_true = tree.flatten(y_pred), tree.flatten(
+                y_true
+            )
+            diff = 0
+            for y_p, y_t in zip(flat_y_pred, flat_y_true):
+                diff += losses.mean_absolute_error(y_t, y_p)
+            return diff
+
+        return loss_fn
+
+    @parameterized.product(
+        type=[tuple, list], anti_type=[list, tuple], weighted=[False, True]
+    )
+    def test_functional_struct_outputs_struct_losses(
+        self, type, anti_type, weighted
```

Comment:
`type` should be a preserved word, and I'm unsure why we need `type` and `anti_type` in this test.

Reply:
This seems a bit complicated to understand and not very intuitive IMO. We might want to document these rules somewhere.
I think we should wait for @fchollet's comment.

Reply:
Changed `anti_type` to `other_type` for better clarity.

Reply:
@james77777778 can you accept the PR if you believe this is resolved? Thanks!

Reply:
LGTM!

I believe the final decision is up to @fchollet

Reply:
Can you approve the changes? The PR shows that it’s missing one reviewer’s approval. Thanks!

Reply:
Unfortunately, I don't have the permission to approve the PR.

Reply:
Oh OK! Have a good day!